### PR TITLE
Hermit Plugin

### PR DESCRIPTION
**Pull request by navyau09**
Hermit plugin activates hermit environment to use and manage packages.

### CODE REVIEW

File: .buildkite/pipeline.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+      - elastic/hermit#v1.0.0
+    agents:
+      image: "docker.elastic.co/ci-agent-images/ubuntu-build-essential"
+  - label: ":sparkles: Buildkite plugin Lint"
+    plugins:
+      - plugin-linter#v3.0.0:
+          id: elastic/hermit
+    agents:
+      provider: "gcp"
+      image: family/ci-ubuntu-2204
```
Comment (jmlrt) [inline .buildkite/pipeline.yaml:14]: is there a way to use a container agent instead of a VM?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  hooks:
+  - id: check-executables-have-shebangs
+  - id: check-json
+  - id: check-merge-conflict
+  - id: check-shebang-scripts-are-executable
+  - id: check-toml
+  - id: check-yaml
+  - id: debug-statements
+  - id: end-of-file-fixer
+#  - id: no-commit-to-branch
```
Comment (jmlrt) [inline .pre-commit-config.yaml]: this should be uncommented

File: .buildkite/pipeline.yaml (hunk 1 above)
Comment (navyau09) [reply .buildkite/pipeline.yaml:14]: Looks like it is using docker command in the [plugin code](https://github.com/buildkite-plugins/plugin-linter-buildkite-plugin/blob/master/hooks/command).

So, I think we cannot use the docker image

**Review by jmlrt — APPROVED** [review]
LGTM, it's a bit sad that we can't use the plugin-linter plugin in a bk container agent. I did a quick try in https://github.com/elastic/hermit-buildkite-plugin/pull/2 but it requires more work and may not be worth it as we shouldn't need to update this codebase (and trigger the BK tests) really often.

**Pull request merged**